src/main/java/com/example/enterprise/MoneyManagerController.java

```diff
@@ -45,7 +45,7 @@ public ResponseEntity fetchAllIncome() {
     }
 
     @GetMapping("/income/{id}/")
-    public ResponseEntity fetchIncomeById(@PathVariable("id") String id) {
+    public ResponseEntity fetchIncomeById(@PathVariable("id") int id) {
```

Comment:
Since your `expenseID` and `incomeID` variables in the DTO has an int object type, therefore you would need this `id` variable to be in int type. To reduce future technical debt (converting from string to int), I changed the variable type to int. 

Here is some more information on technical debt: https://martinfowler.com/bliki/TechnicalDebt.html

Reply:
Are you sure it's a primary key style ID throughout?  If so, great.  If not, it could be a business key, which could be alphanumeric.